crates/hotshot-stake-table/src/mt_based/internal.rs

```diff
@@ -75,35 +91,41 @@ pub struct MerkleProof<K> {
 }
 
 impl<K: Key> MerkleProof<K> {
+    /// Returns the height of the tree
     pub fn tree_height(&self) -> usize {
         self.path.len() - 1
     }
 
+    /// Returns the index of the given key
     pub fn index(&self) -> &usize {
         &self.index
     }
 
+    /// TODO document
```

Comment:
@mrain Can you reply to this with what this function should have as a comment?  We're fixing all our missing comment clippy lints

Reply:
May I submit a PR to this branch fixing these issues? It looks massive.

Reply:
Yes, totally!

Reply:
Thank you :)

Reply:
Check here: https://github.com/EspressoSystems/HotShot/pull/2429